code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -343,8 +349,47 @@
 			force_effect = round(W.force/2)
 		if(prob(10 + force_effect))
 			discipline_slime(user)
+	if(istype(W, /obj/item/storage/bag/bio))
+		var/obj/item/storage/P = W
+		if(!effectmod)
+			to_chat(user, "<span class='warning'>The slime is not currently being mutated.</span>")
+			return
+		var/hasOutput = FALSE //Have we outputted text?
+		var/hasFound = FALSE //Have we found an extract to be added?
+		for(var/obj/item/slime_extract/S in P.contents)
+			if(S.effectmod == effectmod)
+				P.remove_from_storage(S, get_turf(src))
+				qdel(S)
+				applied++
+				hasFound = TRUE
+			if(applied >= SLIME_EXTRACT_CROSSING_REQUIRED)
+				to_chat(user, "<span class='notice'>You feed the slime as many of the extracts from the bag as you can, and it mutates!</span>")
+				playsound(get_turf(src), 'sound/effects/attackblob.ogg', 50, 1)
+				spawn_corecross()
+				hasOutput = TRUE
+				break
+		if(!hasOutput)
+			if(!hasFound)
+				to_chat(user, "<span class='warning'>There are no extracts in the bag that this slime will accept!</span>")
+			else
+				to_chat(user, "<span class='notice'>You feed the slime some extracts from the bag.</span>")
+				playsound(get_turf(src), 'sound/effects/attackblob.ogg', 50, 1)
+		return
 	..()
 
+/mob/living/simple_animal/slime/proc/spawn_corecross()
+	src.visible_message("<span class='danger'>[src] shudders, its mutated core consuming the rest of its body!</span>")
+	playsound(get_turf(src), 'sound/magic/smoke.ogg', 50, 1)
+	var/sanitizedcolour = replacetext(colour, " ", "")
+	var/sanitizedeffect = replacetext(effectmod, "-","")
+	var/corecross = sanitizedeffect + "/" + sanitizedcolour
+	var/crosspath = text2path("/obj/item/slimecross/"+corecross)
```

Comment:
I'd rather you made these properties of the objects and searched through that instead of going by path names.

Reply:
How do you suggest I go about that? Just loop through the subtypes?

Reply:
Yeah jut loop through cached subtypes and compare through initial() so there's no need to instantiate.